crates/test/macros/src/lib.rs

```diff
@@ -140,6 +140,17 @@ fn tests_runner_impl(args: TokenStream) -> parse::Result<TokenStream> {
                 }
             }
         }
+
+        #[cfg(all(test, miri))]
+        #[start]
+        fn main(argc: isize, argv: *const *const u8) -> isize {
+            let ret = __test_runner::main(::core::ptr::null_mut());
```

Comment:
Same here, let's add `unsafe` and the contract for the pointer.

Reply:
This cannot be `unsafe fn`, as it is required to exactly match the expected type of the `#[start]` function.